windows/vtpm_cold_add_remove/vtpm_add_remove_test.yml

```diff
@@ -0,0 +1,84 @@
+# Copyright 2023 VMware, Inc.
+# SPDX-License-Identifier: BSD-2-Clause
+---
+# Current VM has at least one snapshot, e.g., 'BaseSnapshot',
+# adding/removing vTPM device will fail with this message
+# "Cannot change encryption state with virtual machine snapshots present.".
+# So here clone a new VM without snapshot to do vTPM test.
+#
+- name: "Initialize the fact of VM encryption state"
+  ansible.builtin.set_fact:
+    vm_encrypt_state_before: ""
+
+# Check if current VM has vTPM device
+- name: "Get vTPM device on current VM"
+  include_tasks: ../../common/vm_get_device_with_label.yml
+  vars:
+    device_label: 'Virtual TPM'
+
+- name: "Set fact of VM encryption state when VM with vTPM device"
+  ansible.builtin.set_fact:
+    vm_encrypt_state_before: "tpm"
```

Comment:
I suggest to add a new variable to store the tpm existence status, e.g. vm_has_vtpm. 
"vm_encrypt_state_before" is only used for VM encryption status. It'll be more clear and easy understanding.

Reply:
Yeah, that makes sense. Thanks.